src/pmi/pmi.cs

```diff
@@ -339,33 +351,27 @@ public PrepareAll(int firstMethod, bool verbose, bool time) : base(firstMethod,
     {
     }
 
-    public override void StartAssembly(Assembly assembly)
+    public override void StartAssembly(Assembly assembly, string assemblyPath)
     {
-        base.StartAssembly(assembly);
+        base.StartAssembly(assembly, assemblyPath);
         if (_verbose)
         {
             Console.WriteLine($"Prepall for {assemblyName}");
         }
-        pmiFullLogFileName = $"{assemblyName}.pmi";
-        pmiPartialLogFileName = $"{assemblyName}.pmiPartial";
-        markerFileName = $"NextMethodToPrep-{assemblyName}.marker";
+        string assemblyPathAsFile = Util.MapPathToFileName(assemblyPath);
+        pmiFullLogFileName = $"{assemblyPathAsFile}.pmi";
+        pmiPartialLogFileName = $"{assemblyPathAsFile}.pmiPartial";
+        markerFileName = $"NextMethodToPrep-{assemblyPathAsFile}.marker";
     }
 
     public override void AttemptMethod(Type type, MethodBase method)
     {
-        // For now only emit marker files if not in verbose mode.
-        //
-        // When may PMIs are invoked by jit-diff / jit-dasm-pmi we
-        // have trouble with file name contention as the names we
-        // pick are not unique enough. But the names also need to be
-        // predictable by DRIVEALL.
-        //
-        // So as a hack we rely on the fact that DRIVEALL always invokes
-        // PMI in verbose mode, and jit-dasm-pmi in quiet mode.
-        if (_verbose)
-        {
-            WriteAndFlushNextMethodToPrepMarker();
-        }
+        // Many concurrent PMIs are invoked by jit-diff / jit-dasm-pmi.
+        // We need to make sure the file names we choose are unique
+        // as well as being predictable by DRIVEALL. We construct a
+        // filename based on the full pathname of the assembly being
+        // used.
+        WriteAndFlushNextMethodToPrepMarker();
```

Comment:
The rest of this code is to use a munged full path as the marker filename for an assembly. I presume this should be unique enough to satisfy the concern here.

Reply:
Seems like it should be good enough, yeah.